### ISSUE TITLE
cd-extras breaks tab expansion of UNC paths

### ISSUE THREAD

**Issue by sba923** [nickcox/cd-extras#35]
Without `cd-extras` being loaded, in PowerShell 7.1.3, if I enter

````
gci \\pnjnas\p\photos\class
````

and hit the `Tab` key I get:

![image](https://user-images.githubusercontent.com/12860484/112138591-fcc31080-8bd1-11eb-9f83-a8a8230c253b.png)

If `cd-extras` (version 2.2.0) is loaded, hitting the `Tab` key in the same context does nothing, and a (caught) exception is raised:

````
PS C:\Users\<REDACTED>> $error[0] | select *

PSMessageDetails      :
Exception             : System.Management.Automation.MethodInvocationException: Exception calling "CompleteInput" with
                        "3" argument(s): "Object reference not set to an instance of an object."
                         ---> System.NullReferenceException: Object reference not set to an instance of an object.
                           at
                        System.Management.Automation.CompletionCompleters.InvokeScriptArgumentCompleter(ScriptBlock
                        scriptBlock, Object[] argumentsToCompleter, List`1 result)
                           at
                        System.Management.Automation.CompletionCompleters.InvokeScriptArgumentCompleter(ScriptBlock
                        scriptBlock, String commandName, String parameterName, String wordToComplete, CommandAst
                        commandAst, CompletionContext context, List`1 resultList)
                           at System.Management.Automation.CompletionCompleters.NativeCommandArgumentCompletion(String
                        commandName, CompiledCommandParameter parameter, List`1 result, CommandAst commandAst,
                        CompletionContext context, Dictionary`2 boundArguments)
                           at System.Management.Automation.CompletionCompleters.ProcessParameter(String commandName,
                        CommandAst commandAst, CompletionContext context, List`1 result,
                        MergedCompiledCommandParameter parameter, Dictionary`2 boundArguments)
                           at System.Management.Automation.CompletionCompleters.GetArgumentCompletionResultsWithSuccess
                        fulPseudoBinding(CompletionContext context, ArgumentLocation argLocation, CommandAst
                        commandAst)
                           at
                        System.Management.Automation.CompletionCompleters.CompleteCommandArgument(CompletionContext
                        context)
                           at System.Management.Automation.CompletionAnalysis.GetResultForIdentifier(CompletionContext
                        completionContext, Int32& replacementIndex, Int32& replacementLength, Boolean isQuotedString)
                           at System.Management.Automation.CompletionAnalysis.GetResultHelper(CompletionContext
                        completionContext, Int32& replacementIndex, Int32& replacementLength, Boolean isQuotedString)
                           at System.Management.Automation.CompletionAnalysis.GetResults(PowerShell powerShell, Int32&
                        replacementIndex, Int32& replacementLength)
                           at System.Management.Automation.CommandCompletion.CompleteInputImpl(Ast ast, Token[]
                        tokens, IScriptPosition positionOfCursor, Hashtable options)
                           at System.Management.Automation.CommandCompletion.CompleteInput(String input, Int32
                        cursorIndex, Hashtable options)
                           at CallSite.Target(Closure , CallSite , Type , String , Int32 , Hashtable )
                           --- End of inner exception stack trace ---
                           at
                        System.Management.Automation.ExceptionHandlingOps.ConvertToMethodInvocationException(Exception
                        exception, Type typeToThrow, String methodName, Int32 numArgs, MemberInfo memberInfo)
                           at CallSite.Target(Closure , CallSite , Type , String , Int32 , Hashtable )
                           at System.Management.Automation.Interpreter.DynamicInstruction`5.Run(InterpretedFrame frame)
                           at
                        System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame
                        frame)
TargetObject          :
CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
FullyQualifiedErrorId : NullReferenceException
ErrorDetails          :
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at <ScriptBlock><End>, <No file>: line 38
                        at PSConsoleHostReadLine,
                        C:\Users\<REDACTED>\OneDrive\Documents\PowerShell\Modules\PSReadLine\2.2.0\PSReadLine.psm1: line 4
PipelineIterationInfo : {}

````

**Comment by nickcox** [comment]
Thanks very much for finding this. 🙇 

I _think_ [this](https://github.com/nickcox/cd-extras/commit/e38637a4c19fe250782ca72d7d19c0c32b836cb5) fixes it although it's a bit of a pain to write a unit test for.

Feel free to pull master and test against your use case if you have time. Otherwise I will cut a new release in the next few days.

**Comment by sba923** [comment]
Great! The fix seems to work (just downloaded master as a zipfile and used `Import-Module -FullyQualifiedName .\cd-extras\cd-extras.psm1`).

**Comment by nickcox** [comment]
Published as [2.9.4](https://www.powershellgallery.com/packages/cd-extras/2.9.4).

**Comment by sba923** [comment]
Test OK. Good job!
